src/Mutation.tsx

```diff
@@ -0,0 +1,237 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import ApolloClient, {
+  PureQueryOptions,
+  MutationUpdaterFn,
+  ApolloError,
+} from 'apollo-client';
+const invariant = require('invariant');
+import { DocumentNode } from 'graphql';
+const shallowEqual = require('fbjs/lib/shallowEqual');
+
+import { OperationVariables } from './types';
+import { parser, DocumentType } from './parser';
+
+export interface MutationResult<TData = any> {
+  data: TData;
+  error?: ApolloError;
+  loading: boolean;
+}
+
+export interface MutationProps<TData = any, TVariables = OperationVariables> {
+  mutation: DocumentNode;
+  variables?: TVariables;
+  optimisticResponse?: Object;
+  refetchQueries?: string[] | PureQueryOptions[];
```

Comment:
@jbaxleyiii if we want to get rid of the query recycler, should we also get rid of the string array in this option? This would look up the currently watched queries by name, which seems to be the same thing that the deprecated `updateQueries` does.